### ISSUE TITLE
test_cudart_register doesn't work on ROCm

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#45202]
Patch in https://github.com/pytorch/pytorch/pull/45159 and it fails on ROCm with:

```
04:07:51 ======================================================================
04:07:51 FAIL: test_cudart_register (__main__.TestCuda)
04:07:51 ----------------------------------------------------------------------
04:07:51 Traceback (most recent call last):
04:07:51   File "/var/lib/jenkins/.local/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 818, in wrapper
04:07:51     method(*args, **kwargs)
04:07:51   File "test_cuda.py", line 288, in test_cudart_register
04:07:51     self.assertTrue(t.is_pinned())
04:07:51 AssertionError: False is not true
04:07:51 
04:07:51 ----------------------------------------------------------------------
```

I'm guessing there's some behavior difference with how `cudaHostRegister` and the ROCm equivalent works

cc @jeffdaily @sunway513

**Comment by jeffdaily** [comment]
Taking a look now.  Thanks for bringing this to our attention.

**Comment by jeffdaily** [comment]
Looks like the behavior of ROCm runtime after hipHostRegister is called in the test is to change the memory type to hipMemoryTypeDevice, instead of hipMemoryTypeHost.  I need to further confirm with a standalone c++ test case.

**Comment by jeffdaily** [comment]
We have assigned our runtime team to further triage this issue.  The following c++ code has the expected result for ROCm <= 3.3.  ROCm 3.5 and newer returns the memory type as Device.

```C++
#include <iostream>
#include <hip/hip_runtime_api.h>

int main(int argc, char ** argv)
{
    constexpr size_t SIZE = 64;
    float *ptr = new float[SIZE];
    hipPointerAttribute_t attr1;
    hipPointerAttribute_t attr2;
    hipError_t err;

    err = hipPointerGetAttributes(&attr1, ptr);
    if (err != hipSuccess) {
        std::cout << "before register, err " << err << std::endl;
        hipGetLastError();
    }
    else {
        std::cout << "attr1.memoryType " << attr1.memoryType << std::endl;
    }

    err = hipHostRegister(ptr, SIZE, 0);
    if (err != hipSuccess) {
        std::cout << "hipHostRegister, err " << err << std::endl;
        hipGetLastError();
        return -1;
    }

    err = hipPointerGetAttributes(&attr2, ptr);
    if (err != hipSuccess) {
        std::cout << "after register, err " << err << std::endl;
        hipGetLastError();
    }
    else {
        std::cout << "attr2.memoryType " << attr2.memoryType << std::endl;
    }

    return 0;
}
```

**Comment by jeffdaily** [comment]
Will be fixed as part of ROCm 3.10 release.

**Comment by jeffdaily** [comment]
Re-enabled in #52739.
